### ISSUE TITLE
[ValueError] Could not parse version constraint: >-0.10.5 

### ISSUE THREAD

**Issue by nackjicholson** [sdispater/poetry#600]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

## Issue

I have python-language-server installed. I am trying to remove another package with `poetry remove other_package` and during dependency resolution I am receiving this error. The same thing happens during a `poetry lock`.

```
[ValueError]                     
Could not parse version constraint: >-0.10.5  
```

I think it's getting hung up here: https://github.com/palantir/python-language-server/blob/develop/setup.py#L54 on a dependency named `rope`.

Not sure why this is happening now, I've been using poetry and this package for a while.

```
Poetry
======

 * Version: 0.12.6
 * Python:  3.6.6


Virtualenv
==========

 * Python:         3.6.6
 * Implementation: CPython
 * Path:           /Users/nackjicholson/Library/Caches/pypoetry/virtualenvs/source-data-service-py3.6


System
======

 * Platform: darwin
 * OS:       posix
 * Python:   /Users/nackjicholson/.pyenv/versions/3.6.6/Python.framework/Versions/3.6
```

**Comment by jacebrowning** [comment]
I believe the error message is correct in that `>-` is indeed not a valid constraint: https://www.python.org/dev/peps/pep-0440/#version-specifiers

Should this be `>=`?

**Comment by nackjicholson** [comment]
@jacebrowning I thought the same thing. I'm not sure why it ever worked, it very well could just be a typo.

**Comment by nackjicholson** [comment]
I put in this, because I think that must have been a typo: https://github.com/palantir/python-language-server/pull/448

I'm still a bit confused why this had been working for me up until today.

**Comment by nackjicholson** [comment]
@jacebrowning I did `poetry self:update 0.12.4` in order to go back to the version of poetry I had prior. Installing python-language-server worked just fine.

I'm at a loss for what is going on, but I think I will keep myself of `0.12.4` for the time being until @sdispater weighs in on this.

**Comment by sdispater** [comment]
@nackjicholson Could you execute `poetry lock` in debug mode (`-vvv`) and post the result here?

**Comment by nackjicholson** [comment]
```
Using virtualenv: /Users/nackjicholson/Library/Caches/pypoetry/virtualenvs/source-data-service-py3.6
Updating dependencies
Resolving dependencies...
   1: fact: source-data-service is 0.3.0
   1: derived: source-data-service
   1: fact: source-data-service depends on flask (^1.0)
   1: fact: source-data-service depends on click (^6.7)
   1: fact: source-data-service depends on boto3 (^1.7)
   1: fact: source-data-service depends on happybase (^1.1)
   1: fact: source-data-service depends on flask-env (^2.0)
   1: fact: source-data-service depends on cerberus (^1.2)
   1: fact: source-data-service depends on PyYAML (^3.13)
   1: fact: source-data-service depends on psycopg2 (^2.7)
   1: fact: source-data-service depends on cl-flask (^1.0.0)
   1: fact: source-data-service depends on clutilities (^3.1)
   1: fact: source-data-service depends on numpy (^1.15)
   1: fact: source-data-service depends on pandas (^0.23.4)
   1: fact: source-data-service depends on pendulum (^2.0)
   1: fact: source-data-service depends on requests (^2.19)
   1: fact: source-data-service depends on cl-boto3 (^1.0)
   1: fact: source-data-service depends on python-json-logger (^0.1.9)
   1: fact: source-data-service depends on python-dotenv (^0.9.1)
   1: fact: source-data-service depends on pytest (^3.7)
   1: fact: source-data-service depends on pytest-cov (^2.5)
   1: fact: source-data-service depends on pytest-postgresql (^1.3)
   1: fact: source-data-service depends on happybase-mock (^0.10.0)
   1: fact: source-data-service depends on moto (^1.3)
   1: fact: source-data-service depends on jinja2-cli (^0.6.0)
   1: fact: source-data-service depends on black (^18.3-alpha.0)
   1: fact: source-data-service depends on flake8 (^3.5)
   1: fact: source-data-service depends on python-language-server (^0.21.2)
   1: fact: source-data-service depends on pyls-black (^0.3.0)
   1: selecting source-data-service (0.3.0)
   1: derived: pyls-black (^0.3.0)
   1: derived: python-language-server (^0.21.2)
   1: derived: flake8 (^3.5)
   1: derived: black (^18.3-alpha.0)
   1: derived: jinja2-cli (^0.6.0)
   1: derived: moto (^1.3)
   1: derived: happybase-mock (^0.10.0)
   1: derived: pytest-postgresql (^1.3)
   1: derived: pytest-cov (^2.5)
   1: derived: pytest (^3.7)
   1: derived: python-dotenv (^0.9.1)
   1: derived: python-json-logger (^0.1.9)
   1: derived: cl-boto3 (^1.0)
   1: derived: requests (^2.19)
   1: derived: pendulum (^2.0)
   1: derived: pandas (^0.23.4)
   1: derived: numpy (^1.15)
   1: derived: clutilities (^3.1)
   1: derived: cl-flask (^1.0.0)
   1: derived: psycopg2 (^2.7)
   1: derived: PyYAML (^3.13)
   1: derived: cerberus (^1.2)
   1: derived: flask-env (^2.0)
   1: derived: happybase (^1.1)
   1: derived: boto3 (^1.7)
   1: derived: click (^6.7)
   1: derived: flask (^1.0)
pypicloud: 1 packages found for pyls-black >=0.3.0,<0.4.0
pypicloud: The cache for pyls-black 0.3.0 is outdated. Refreshing.
pypicloud: Downloading wheel: pyls_black-0.3.0-py3-none-any.whl#sha256=af4bfd73caeefd31f8035c20d30655fe26fa64e6e802d3c62e92465c13dc5e69
   1: fact: pyls-black (0.3.0) depends on python-language-server (*)
   1: fact: pyls-black (0.3.0) depends on black (>=18.9b0)
   1: fact: pyls-black (0.3.0) depends on toml (*)
   1: selecting pyls-black (0.3.0)
   1: derived: toml (*)
   1: derived: black (>=18.9b0)
pypicloud: 1 packages found for python-language-server >=0.21.2,<0.22.0
pypicloud: The cache for python-language-server 0.21.2 is outdated. Refreshing.
pypicloud: Downloading sdist: python-language-server-0.21.2.tar.gz#sha256=fa9162acb1402b807132d7288b7f521db2bd666d63505d8a4d9464d4b8488c52
   1: Version solving took 1.851 seconds.
   1: Tried 1 solutions.
                                                
[ValueError]                     
Could not parse version constraint: >-0.10.5  
                                                
Exception trace:
 /Users/nackjicholson/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /Users/nackjicholson/.poetry/lib/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /Users/nackjicholson/.poetry/lib/poetry/_vendor/py3.6/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /Users/nackjicholson/.poetry/lib/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /Users/nackjicholson/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /Users/nackjicholson/.poetry/lib/poetry/_vendor/py3.6/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /Users/nackjicholson/.poetry/lib/poetry/console/commands/lock.py in handle() at line 33
   return installer.run()
 /Users/nackjicholson/.poetry/lib/poetry/installation/installer.py in run() at line 76
   self._do_install(local_repo)
 /Users/nackjicholson/.poetry/lib/poetry/installation/installer.py in _do_install() at line 158
   ops = solver.solve(use_latest=self._whitelist)
 /Users/nackjicholson/.poetry/lib/poetry/puzzle/solver.py in solve() at line 38
   packages, depths = self._solve(use_latest=use_latest)
 /Users/nackjicholson/.poetry/lib/poetry/puzzle/solver.py in _solve() at line 171
   self._package, self._provider, locked=locked, use_latest=use_latest
 /Users/nackjicholson/.poetry/lib/poetry/mixology/__init__.py in resolve_version() at line 7
   return solver.solve()
 /Users/nackjicholson/.poetry/lib/poetry/mixology/version_solver.py in solve() at line 79
   next = self._choose_package_version()
 /Users/nackjicholson/.poetry/lib/poetry/mixology/version_solver.py in _choose_package_version() at line 378
   version = self._provider.complete_package(version)
 /Users/nackjicholson/.poetry/lib/poetry/puzzle/provider.py in complete_package() at line 432
   package.name, package.version.text, extras=package.requires_extras
 /Users/nackjicholson/.poetry/lib/poetry/repositories/pool.py in package() at line 67
   package = repository.package(name, version, extras=extras)
 /Users/nackjicholson/.poetry/lib/poetry/repositories/legacy_repository.py in package() at line 262
   dependency = dependency_from_pep_508(req)
 /Users/nackjicholson/.poetry/lib/poetry/packages/__init__.py in dependency_from_pep_508() at line 34
   req = Requirement(name)
 /Users/nackjicholson/.poetry/lib/poetry/version/requirements.py in __init__() at line 222
   self.constraint = parse_constraint(constraint)
 /Users/nackjicholson/.poetry/lib/poetry/semver/__init__.py in parse_constraint() at line 31
   constraint_objects.append(parse_single_constraint(and_constraints[0]))
 /Users/nackjicholson/.poetry/lib/poetry/semver/__init__.py in parse_single_constraint() at line 164
   raise ValueError("Could not parse version constraint: {}".format(constraint))

lock
```

Here  ya go!

**Comment by sdispater** [comment]
Okay I made a change in commit 9c63f78 that will make the legacy repositories behave like when searching from PyPI, i.e. drop the invalid constraint.

This commit also fixes the fact that extras are simply not picked up for legacy repositories.

**Comment by sdispater** [comment]
I just released version `0.12.7` with this fix included.
